src/Exception/ConfigurationException.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types=1);
+/*
+ * This file is part of sebastian/diff.
+ *
+ * (c) Sebastian Bergmann <sebastian@phpunit.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SebastianBergmann\Diff;
+
+final class ConfigurationException extends InvalidArgumentException
+{
+    /**
+     * @param string         $option
+     * @param string         $expected
+     * @param mixed          $value
+     * @param int            $code
+     * @param Exception|null $previous
+     */
+    public function __construct(
+        string $option,
+        string $expected,
+        $value,
+        int $code = 0,
+        Exception $previous = null
```

Comment:
this is violation of parent.
parent expects `\Exception`, here, we expect `Exception`.
the thing is, that if one raise `new ConfigurationException('A', 'B', 'C', 0, new \RuntimeException())`, if will crash as `Exception` is local interface that php build-in exceptions are not implementing.

Also, please add test case will all parameters provided, it is missing currently

Reply:
nice catch, thanks :)
fixed (I hope)